src/gatherers/screenshots.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Gather = require('./gather');
+const DevtoolsTimelineModel = require('../lib/traces/devtools-timeline-model');
+const fs = require('fs');
+const log = require('../lib/log.js');
+
+const starthtml = `<body>
+<script>
+var shots = `;
+const endhtml = `
+shots.forEach(s => {
+  var i = document.createElement('img');
+  i.src = s.datauri;
+  i.title = s.timestamp;
+  document.body.appendChild(i);
+});
+</script>
+`;
+
+class ScreenshotFilmstrip extends Gather {
+
+  fetchScreenshot(frame) {
+    return frame
+      .imageDataPromise()
+      .then(data => Promise.resolve('data:image/jpg;base64,' + data));
+  }
+
+  /**
+   * @param {!Array<!Object>} traceData
+   * @return {!Promise}
+  */
+  getScreenshots(traceData) {
+    const model = new DevtoolsTimelineModel(traceData);
+    const filmStripFrames = model.filmStripModel().frames();
+
+    const frameFetches = filmStripFrames.map(frame => this.fetchScreenshot(frame));
+    return Promise.all(frameFetches).then(images => {
+      return filmStripFrames.map((frame, i) => ({
+        timestamp: frame.timestamp,
+        datauri: images[i]
+      }));
+    });
+  }
+
+  postProfiling(options, tracingData) {
+    return this.getScreenshots(tracingData.traceContents).then(screenshots => {
+      if (options.flags.saveScreenshots) {
```

Comment:
This should live outside the gatherer. We already have code out for saving a trace file, it should live right next to that, and ideally the code that bakes the images should use Handlebars for consistency with the report.